src-built-in/components/userPreferences/src/components/general/Toolbar.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+export default class Toolbar extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			monitor: "all"
+		};
+		this.setMonitor = this.setMonitor.bind(this);
+		this.restartApplication = this.restartApplication.bind(this);
+	}
+
+    /**
+     * Sets the meridiem for the time (AM or PM). Afterwards, this calls setHour to handle any necessary conversions to military time.
+     * @param {event} e
+     */
+	setMonitor(e) {
+		let monitor = e.target.value;
+		this.setState({
+			monitor: monitor
+		});
+		FSBL.Clients.ConfigClient.setPreference({ field: "finsemble.components.Toolbar.window.monitor", value: monitor });
+		// There's no need to override spawnOnAllMonitors. Setting monitor to anything other than undefined will automatically override that value. Setting monitor="all" is equivalent.
+	}
+
+	restartApplication() {
+		FSBL.restartApplication();
+	}
+
+    /**
+     * Add listener on the store. When the preferences field changes, we change our local state.
+     * Also, get the intiial state from the store.
+     */
+	componentDidMount() {
+		FSBL.Clients.ConfigClient.getValue("finsemble.components.Toolbar.window.monitor", (err, value) => {
+			if (!value) value = "all";
+			this.setState({
+				monitor: value
+			});
+		});
+	}
+
+	componentWillUnmount() {
+	}
+
+	render() {
+		return <div className="complex-menu-content-row">
+			<div>
+				<span>Toolbar Monitor (requires restart)</span>
+				<select style={{ margin: "0px 10px" }} onChange={this.setMonitor} value={this.state.monitor}>
```

Comment:
This should get all monitors from the launcher and just 

```
monitors.map((monitor, i)=>{
    return <option value={i}>{i}</option>
});
```

Don't know if you did this just for protyping, but it should definitely get the monitors from the launcher. Think it's getMonitorInfoAll

Reply:
The reason I didn't do that is because you may move to another system with fewer monitors. The actual value that is set for the monitor will no longer match, it won't be an option. You then get start getting into writing a lot of code to deal with a minor issue. Considering that we plan on moving to a "perfect toolbar", and that this code is essentially temporary, the hard coded values seemed reasonable.